subcommand/health-sync/command_test.go

```diff
@@ -0,0 +1,261 @@
+	err := consulClient.Agent().ServiceRegister(&api.AgentServiceRegistration{
+		ID:   serviceID,
+		Name: serviceName,
+		Port: 1000,
+		Meta: map[string]string{
+			"task-id":  taskID,
+			"task-arn": taskARN,
+			"source":   "consul-ecs",
+		},
+	})
+	require.NoError(t, err)
+
+	for _, container := range containers {
+		if !container.essential {
+			continue
+		}
+
+		err = consulClient.Agent().CheckRegister(&api.AgentCheckRegistration{
+			Name:      makeCheckID(serviceID, container.name),
+			ServiceID: serviceID,
+			AgentServiceCheck: api.AgentServiceCheck{
+				TTL: "10000h",
+			},
+		})
+		require.NoError(t, err)
+	}
+
+}
+
+func assertHealthChecks(t *testing.T, consulClient *api.Client, containers []minimalContainerMetadata) {
+	filter := fmt.Sprintf(`ServiceID == "%s"`, serviceID)
+	checks, err := consulClient.Agent().ChecksWithFilter(filter)
+	require.NoError(t, err)
+	for _, container := range containers {
+		checkID := makeCheckID(serviceID, container.name)
+		check, ok := checks[checkID]
+
+		if !container.essential {
+			require.False(t, ok)
+			continue
+		}
+
+		require.True(t, ok)
+		require.Equal(t, ecsHealthToConsulHealth(container.status), check.Status)
+	}
+}
+
+var cases map[string][]minimalContainerMetadata = map[string][]minimalContainerMetadata{
+	"one essential healthy container": {{
+		name:      "container1",
+		status:    "HEALTHY",
+		essential: true,
+	},
+	},
+	"two essential containers that are healthy": {{
+		name:      "container1",
+		status:    "HEALTHY",
+		essential: true,
+	},
+		{
+			name:      "container2",
+			status:    "HEALTHY",
+			essential: true,
+		},
+	},
+	"two essential containers where one is unhealthy": {{
+		name:      "container1",
+		status:    "HEALTHY",
+		essential: true,
+	},
+		{
+			name:      "container2",
+			status:    "UNHEALTHY",
+			essential: true,
+		},
+	},
+	"one essential container and one non-essential": {{
+		name:      "container1",
+		status:    "HEALTHY",
+		essential: true,
+	},
+		{
+			name:      "container2",
+			status:    "UNHEALTHY",
+			essential: false,
+		},
+	},
+}
+
+func TestRunWithContainerNames(t *testing.T) {
+	for name, containers := range cases {
+		t.Run(name, func(t *testing.T) {
+
+			server, err := testutil.NewTestServerConfigT(t, nil)
+			require.NoError(t, err)
+			t.Cleanup(func() {
+				_ = server.Stop()
+				os.Unsetenv("CONSUL_HTTP_ADDR")
+			})
+
+			consulClient, err := api.NewClient(&api.Config{Address: server.HTTPAddr})
+			require.NoError(t, err)
+			// We need to set this so that consul connect envoy -bootstrap will talk to the right agent.
+			os.Setenv("CONSUL_HTTP_ADDR", server.HTTPAddr)
+
+			setupServiceAndChecks(t, consulClient, containers)
+
+			essentialContainers := []string{}
+			for _, container := range containers {
+				if container.essential {
+					essentialContainers = append(essentialContainers, container.name)
+				}
+			}
```

Comment:
I'm wondering if we actually need this logic around essential containers in this test? 

My read of the code is that `health-sync -container-names A,B,C` would sync status of containers A, B, and, C - regardless of whether they are essential (right?)

(We agreed that it makes sense for Terraform to pass only the essential containers to health-sync, but `health-sync` does not need to know/care about that.)

Reply:
This is testing the case where there is a container that isn't essential. It [doesn't create a check for non-essential containers](https://github.com/hashicorp/consul-ecs/pull/33/files#diff-d2e144a88d61b2f20b2e5ba950a301a0b6d5328c70979b7713d66ffec95a5b6fR69-R71) and that ensures we aren't trying to set [a TTL status for a non-essential container](https://github.com/hashicorp/consul-ecs/pull/33/files#diff-1d2a3112468dcca20d98917f20cd62cf528be35db0af6f7e1274d599c4bb6a12R179) (which would fail). This test code is a bit more complicated than I would optimally want tests to be, but I think it is testing a legitimate case. Does this make sense to you?

Reply:
Oh, okay. I think I understand.

1. You're using that `essential` field purely for test setup, to control which containers are prepared with a corresponding TTL check?
2. My read of the code is that the `health-sync` command will attempt to sync status for all container names given, regardless of whether the containers are essential. Is that intentional? (It seems like the right behavior to me)

Assuming "yes" to both of those, then this is good with me.

Reply:
Yes and yes. Thanks for thinking through this with me 😄